File: KoKoslan/src/java/ast/KoKoAst.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.java.ast;

import java.io.*;
import kokoslan.java.eval.*;

public interface KoKoAst{
   default void genCode(){
      genCode(System.out);
   }
   default void genCode(PrintStream out){
   }
   KoKoValue eval(KoKoContext ctx);
   
}

File: KoKoslan/src/kotlin/ast/KoKoAst.kt
/** 
   Andrey Arguedas Espinoza
   Daniela Armas Sanchez
   Michael Chen Wang
   Kimberly Olivas Delgado
*/

package kokoslan.kt.ast

import java.io.*
import kokoslan.kt.eval.*
import kokoslan.kt.exception.*

interface KoKoAst {
   fun genCode() {
      genCode(System.out)
   }
   
   fun genCode(Out : PrintStream)
   
   fun eval(ctx : KoKoContext) : KoKoValue?
   
}